src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -595,17 +597,17 @@ public static void ReconcileSlotNames<T>(IHostEnvironment env, IDataView[] views
         /// data view, with the union of the key values as the new key values. For each data view, the value in the output column is the value
         /// corresponding to the key value in the original column.
         /// </summary>
-        public static void ReconcileKeyValues(IHostEnvironment env, IDataView[] views, string columnName)
+        public static void ReconcileKeyValues(IHostEnvironment env, IDataView[] views, string columnName, ColumnType keyValueType)
         {
             Contracts.CheckNonEmpty(views, nameof(views));
             Contracts.CheckNonEmpty(columnName, nameof(columnName));
 
             var dvCount = views.Length;
 
-            Dictionary<DvText, int> keyNames;
-            int[] indices;
             // Create mappings from the original key types to the reconciled key type.
-            var keyValueMappers = MapKeys(views.Select(view => view.Schema).ToArray(), columnName, false, out indices, out keyNames);
+            var indices = new int[dvCount];
+            var keyNames = new Dictionary<DvText, int>();
+            var keyValueMappers = Utils.MarshalInvoke(MapKeys<int>, keyValueType.RawType, views.Select(view => view.Schema).ToArray(), columnName, false, indices, keyNames);
```

Comment:
>Utils.MarshalInvoke [](start = 34, length = 19)

can you add comment what you use MarshalInvoke to pass keyValueType.RawType as generic type to function?